modules/460-log-shipper/hooks/internal/vector/destination/loki.go

```diff
@@ -112,6 +118,33 @@ func NewLoki(name string, cspec v1alpha1.ClusterLogDestinationSpec) *Loki {
 		tls.VerifyHostname = *spec.TLS.VerifyHostname
 	}
 
+	endpoint, err := url.Parse(spec.Endpoint)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse loki endpoint '%s'", spec.Endpoint)
+	}
+
+	lokiAddr, err := net.ResolveTCPAddr("tcp", endpoint.Host)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve ip address for loki endpoint '%s'", spec.Endpoint)
+	}
+
+	deckhouseLokiAddr, err := net.ResolveTCPAddr("tcp", d8LokiPodAddress)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve ip address for deckhouse loki pod address '%s'", d8LokiPodAddress)
+	}
```

Comment:
This is not good because the address may never be resolved in case an invalid address is specified, which will block the main queue. There may also be a problem if the loki pod is in crshloopbackopff state.

Reply:
I started thinking that the Kubernetes resolution is not reliable enough for a hook to make a decision. Let's just stick to the endpoint itself and do not try to resolve it.

You need to check that enpoint is in all possible range

```
http://loki.d8-monitoring:3000
http://loki.d8-monitoring.:3000
http://loki.d8-monitoring.svc.<domain>:3000
http://loki.d8-monitoring.svc.<domain>.:3000
http://<cluster-ip>:3000
```

Reply:
done